init.go

```diff
@@ -1,21 +1,82 @@
 package allure
 
 import (
+	"fmt"
 	"github.com/jtolds/gls"
+	"io"
+	"log"
+	"os"
+	"strings"
 )
 
 var (
-	ctxMgr     *gls.ContextManager
-	wsd        string
-	resultPath string
+	ctxMgr *gls.ContextManager
+	wsd    string
 )
 
 const (
 	resultsPathEnvKey = "ALLURE_RESULTS_PATH"
 	wsPathEnvKey      = "ALLURE_WORKSPACE_PATH"
+	envFileKey        = "ALLURE_ENVIRONMENT_FILE_PATH"
 	nodeKey           = "current_step_container"
 )
 
+var resultsPath string
+
 func init() {
 	ctxMgr = gls.NewContextManager()
+	if envFilePath := os.Getenv(envFileKey); envFilePath != "" {
+		envFilesStrings := strings.Split(envFilePath, "/")
+
+		resultsPathEnv := os.Getenv(resultsPathEnvKey)
```

Comment:
This should not happen in an init func since this environment variable could be set in the upstream project code, and would not be set at the init time. See https://github.com/dailymotion/allure-go/pull/7#discussion_r342481765